src/java/org/apache/cassandra/db/CounterMutation.java

```diff
@@ -172,7 +205,7 @@ public Mutation applyCounterMutation() throws WriteTimeoutException
         Mutation.PartitionUpdateCollector resultBuilder = new Mutation.PartitionUpdateCollector(getKeyspaceName(), key());
         Keyspace keyspace = Keyspace.open(getKeyspaceName());
 
-        List<Lock> locks = new ArrayList<>();
+        ArrayList<Lock> locks = new ArrayList<>();
         Tracing.trace("Acquiring counter locks");
```

Comment:
nit: this is almost useless, in this critical path we can remove it, I don't think it adds value

Reply:
In general, the code should be written with two stakeholders in mind:
1. the actual executor, i.e. the computer
2. the subsequent reader of the code
There is no reason to make life harder for either if it brings no value. I'd also argue that except for critical paths, we care more for 2 than 1.

As you pointed out, the computer doesn't really care about the type specifier in this particular instance.
However, from the perspective of the human who reads the code, the benefit is apparent. The reverse iteration that happens during unlocking (I don't know a more elegant way than the one used) uses the `get(k)` method of a list. 
The next person seeing `List::get` will think "shit, do we have an O(n^2) here?" and will waste time and mental energy checking that.
Declaring the type as `ArrayList` makes it obvious that these calls are O(1).

Reply:
this comment was about the Tracing.trace statement, we can remove it


Regarding the type of variable, I understand your point.
I am not super conviced that using ArrayList as the type explains to the reader your motivation.
If we want to point this out to the future reader then I would add a comment.
I don't think we have to do anything now.
I am fine with keeping ArrayList as you are doing now.